s3store/s3store.go

```diff
@@ -547,3 +560,40 @@ func isAwsError(err error, code string) bool {
 	}
 	return false
 }
+
+func (store S3Store) CalcOptimalPartSize(size int64) (optimalPartSize int64, err error) {
+	var errorstrings []string
+
+	switch {
+	// When upload is smaller or equal MinPartSize, we upload in just one part.
+	case size <= store.MinPartSize:
+		optimalPartSize = store.MinPartSize
+	// When we need 9999 parts or less with MinPartSize.
+	case size/store.MinPartSize < store.MaxMultipartParts:
```

Comment:
Could we remove the integer division here and test for `size < MinPartSize*MaxMultipartParts` instead? I don't think this makes a difference here but I would like to avoid integer division whenever possible since it does not behave intuitively first :)

Reply:
We can do that. My emphasis lies on readability and understandability and this change does rather further them then harm them.